vyper/ast/expansion.py

```diff
@@ -38,12 +38,19 @@ def generate_public_variable_getters(vyper_module: vy_ast.Module) -> None:
         # use the annotation node as a base to build the input args and return type
         # starting with `args[0]` to remove the surrounding `public()` call`
         annotation = copy.copy(node.annotation.args[0])
-
-        # the base return statement is an `Attribute` node, e.g. `self.<var_name>`
-        # for each input type we wrap it in a `Subscript` to access a specific member
-        return_stmt: vy_ast.VyperNode = vy_ast.Attribute(
-            value=vy_ast.Name(id="self"), attr=func_type.name
-        )
+        # if we have a mutability declaration after a visibility one, skip over
+        if node.is_constant or node.is_immutable:
+            annotation = annotation.args[0]
+
+        # constants just return a value
+        if node.is_constant:
+            return_stmt: vy_ast.VyperNode = node.value
+        else:
```

Comment:
i think the immutables thing will be fixed if you just return a `vy_ast.Name(id=func_type.name)` instead of returning a whole `self.<func_type.name>` node.